### PR TITLE
Replace company edit form with newer macros

### PR DESCRIPTION
**Pull request by teneightfive**
This uses the newer pattern for form macros and removes the last use of the older deprecated form macros.

It still doesn't use the new form builder method but allows the last remaining form macros to be removed.

## Before
localhost_3001_companies_cf557dfb-44f3-4afc-8bee-a4df4afddc98_edit

## After
localhost_3001_companies_cf557dfb-44f3-4afc-8bee-a4df4afddc98_edit 3

### CODE REVIEW

File: src/apps/companies/middleware/form.js [hunk 1]
```diff
@@ -4,10 +4,23 @@ const metadataRepository = require('../../../lib/metadata')
 const { companyDetailsLabels, hqLabels, chDetailsLabels } = require('../labels')
 const companyFormService = require('../services/form')
 const { transformCompanyResponseToForm } = require('../transformers')
+const { transformObjectToOption } = require('../../transformers')
 
 const chDetailsDisplayOrder = ['name', 'company_number', 'registered_address', 'business_type', 'company_status', 'incorporation_date', 'sic_code']
 
 function populateForm (req, res, next) {
+  const headquarterOptions = metadataRepository.headquarterOptions.map(transformObjectToOption).map(option => {
```
Comment (tyom) [inline src/apps/companies/middleware/form.js]: Maybe start each `map` on the new line for easier scanning?

File: src/apps/companies/middleware/form.js [hunk 2]
```diff
@@ -22,17 +35,20 @@ function populateForm (req, res, next) {
...
     chDetailsDisplayOrder,
-    regionOptions: metadataRepository.regionOptions,
-    sectorOptions: metadataRepository.sectorOptions,
-    employeeOptions: metadataRepository.employeeOptions,
-    turnoverOptions: metadataRepository.turnoverOptions,
-    headquarterOptions: metadataRepository.headquarterOptions,
-    countryOptions: metadataRepository.countryOptions,
+    headquarterOptions,
+    regionOptions: metadataRepository.regionOptions.map(transformObjectToOption),
+    sectorOptions: metadataRepository.sectorOptions.map(transformObjectToOption),
```
Comment (tyom) [inline src/apps/companies/middleware/form.js]: A lot of these are defined in `globalFields` in `apps/macros.js`.

Comment (tyom) [reply src/apps/companies/middleware/form.js]: Oh, I just realised they are not proper fields for form builder yet.

Comment (teneightfive) [reply src/apps/companies/middleware/form.js]: >Oh, I just realised they are not proper fields for form builder yet.

Yep, as mentioned in the description this is the first step of moving to form builder. This change is to get rid of legacy macros.

**Pull request merged**